### DISCUSSION TITLE
Build without X11 for qt6

### DISCUSSION THREAD

**Discussion by oniGino** [tsujan/Kvantum#948] [Ideas]
Looks like you have some effort towards Native Wayland environments but still call some X11 libraries unconditionally.

In cases where qt6base is build without X11 support numerous compiletime errors regarding

`blur/blurhelper.cpp:52:74: error: no member named 'QX11Application'`
`style/themeconfig/ThemeConfig.cpp:595:74: error: no member named 'QX11Application' in namespace 'QNativeInterface'`

as a some examples..

At a glance I think if defined could be adjusted to a nested structure where

```
#if (QT_VERSION < QT_VERSION_CHECK(6,0,0))
...
-#else
+#elif defined Q_WS_X11
...
```

or something like that
or maybe don't use Q_WS_X11 at all?

Thanks for all your work

**Comment by tsujan** [comment]
> In cases where qt6base is build without X11 support

Why should someone build qt6base without X11 support?!

**Reply by 0xIO32** [reply]
Less dependencies, which you don't need
